src/main/java/hudson/remoting/Util.java

```diff
@@ -131,4 +126,32 @@ static URLConnection openURLConnection(URL url) throws IOException {
         }
         return con;
     }
+
+    static InetSocketAddress getResolvedHttpProxyAddress(String host, int port) throws IOException {
+        InetSocketAddress targetAddress = null;
+        Iterator<Proxy> proxies = ProxySelector.getDefault().select(URI.create(String.format("http://%s:%d", host, port))).iterator();
+        while (targetAddress == null && proxies.hasNext()) {
+            Proxy proxy = proxies.next();
+            if(proxy.type() == Proxy.Type.HTTP) {
+                if (!(proxy.address() instanceof InetSocketAddress))
+                    throw new IOException("Unsupported proxy address type " + proxy.address().getClass());
```

Comment:
Should this really `throw` rather than log? It's looking for a suitable `Proxy` after all.